components/feature/push/src/main/java/mozilla/components/feature/push/AutoPushFeature.kt

```diff
@@ -300,7 +306,7 @@ class AutoPushFeature(
     }
 
     @VisibleForTesting(otherwise = VisibleForTesting.PRIVATE)
-    internal fun tryVerifySubscriptions() {
+    internal fun tryVerifySubscriptions() = coroutineScope.launch {
```

Comment:
Hm I don't see `tryVerifySubscriptions` doing anything that would need to run on the scope except `verifyActiveSubscriptions` which is being launched on the scope already. Is this additional launch necessary?

Reply:
It needs the scope for the `shouldVerifyNow` method which touches IO when checking prefs.

> Is this additional launch necessary?

Yeah, I wasn't super keen for that nested scope call either.

Reply:
Ah, I see the getter of `prefLastVerified` touches preferences. I missed that part. We could inline `verifyActiveSubscriptions` or just make it a `suspend` function to prevent the nested launch. But good either way from my perspective.

Reply:
Making it a suspend method is a good idea since it's internal. 🙂